ogc/wms-layer-url/src/main/java/com/esri/samples/wms_layer_url/WmsLayerUrlSample.java

```diff
@@ -70,22 +70,24 @@ public void start(Stage stage) {
       // start zoomed in over the US
       mapView.setViewpointGeometryAsync(new Envelope(-19195297.778679, 512343.939994, -3620418.579987, 8658913.035426, 0.0, 0.0, SpatialReferences.getWebMercator()));
 
-      // show a progress indicator while the layer loads
-      ProgressIndicator progressIndicator = new ProgressIndicator();
-      progressIndicator.setProgress(ProgressIndicator.INDETERMINATE_PROGRESS);
+      // create a progress indicator and set its size
+      var progressIndicator = new ProgressIndicator();
       progressIndicator.setMaxSize(25, 25);
 
       // create a WMS layer
       List<String> wmsLayerNames = Collections.singletonList("1");
       String url = "https://nowcoast.noaa.gov/arcgis/services/nowcoast/radar_meteo_imagery_nexrad_time/MapServer/WMSServer?request=GetCapabilities&service=WMS";
       wmsLayer = new WmsLayer(url, wmsLayerNames);
+
+      // show a progress indicator while the layer loads
+      progressIndicator.visibleProperty().bind(wmsLayer.loadStatusProperty().isEqualTo(LoadStatus.LOADING));
+
       // load the layer and add it as an operational layer
-      wmsLayer.addDoneLoadingListener(() -> {
-        if (wmsLayer.getLoadStatus() != LoadStatus.LOADED) {
-          wmsLayer.getLoadError().printStackTrace();
+      wmsLayer.loadStatusProperty().addListener((observable, oldValue, newValue) -> {
```

Comment:
@Rachael-E The property listener should work just as well as the done loading BUT you have to remember it will fire for each state so you might need to deal with intermediate `LOADING` state where as done loading will only happen for `LOADED` or `FAILED` to load.